src/Pact/Types/ChainMeta.hs

```diff
@@ -120,15 +160,31 @@ makeLenses ''PublicMeta
 
 instance Default PublicMeta where def = PublicMeta "" "" 0 0 0 0
 
+instance Arbitrary PublicMeta where
+  arbitrary = PublicMeta
+    <$> arbitrary
+    <*> arbitrary
+    <*> arbitrary
+    <*> arbitrary
+    <*> arbitrary
+    <*> arbitrary
+
+publicMetaProperties :: JsonProperties PublicMeta
+publicMetaProperties o =
```

Comment:
Can we automate these `JsonProperties` function using Generics? This seems like something whether both naming and ordering and very significant, and prone to future mistakes. For example, if a field is added to the structure, you won't get a warning here that the property is missing that field. If we don't automate, we would need tests for every one of these functions to ensure total and correctly ordered representation.

Reply:
Stable serialization functions shouldn't be derived generically from internal datatypes. This is what we used to do and what got us here. With generically derived instances when a programmer makes a change to the data type (usually motivated by some algorithmic needs), they would break the API without noticing. Even worse, if API clients, that are used in tests are also derived, it won't show up in tests, but everybody else is broken.

An alternative would be to declare dedicated data types just for serialization/API purposes and define a mapping from internal types to the API types. But I think, the overhead and benefits are often similar to defining dedicated serialization functions.